### PR TITLE
Update OError to v3

### PR DESCRIPTION
**Pull request by mserranom**
### Description

Updates the service to use [OError v3](https://github.com/overleaf/o-error).

Before merging we need to update  `logger-sharelatex` module compatibility with `OError` v3 and bumpt its version here.

#### Potential Impact

Logging and error recovery scenarios.

#### Manual Testing Performed

- [x] service in dev-environment continues working as usual
- [ ] stack traces are delivered according to OError v3 spec

#### Who Needs to Know?

@jdleesmiller @warncke

### CODE REVIEW

File: app/js/ASpell.js [hunk 1]
```diff
@@ -27,21 +28,22 @@ try {
   const oldCache = fs.readFileSync(cacheFsPath)
   cache.load(JSON.parse(oldCache))
 } catch (error) {
-  const err = error
-  logger.log({ err, cacheFsPath }, 'could not load the cache file')
+  logger.log(
+    OError.tag(error, 'could not load the cache file', { cacheFsPath })
+  )
```
[the remark this answers is not in the document]
Comment (mserranom) [reply app/js/ASpell.js:33]: Haven't tested yet (hence the unchecked box in the description), still need to update `logger-sharelatex` to `OError` v3

However I expected it to work with the custom serialiser https://github.com/overleaf/logger-module/blob/master/logging-manager.js#L8-L20

> bunyan will only use the error serializer if the first argument has an err property

I can't find that restriction in the bunyan link, where is it indicated?

[the remark this answers is not in the document]
Comment (emcsween) [reply app/js/ASpell.js:33]: It's not super clear, but it's explained [a bit higher up in the README](https://github.com/trentm/node-bunyan#serializers):

> A logger instance can have a serializers mapping of log record field name ("req" in this example) to a serializer function. When creating the log record, Bunyan will call the serializer function for top-level fields of that name.

[the remark this answers is not in the document]
Comment (mserranom) [reply app/js/ASpell.js:33]: Run a few tests, results a bit unexpected, will try to reproduce a general case.
```
logger.log(new OError('message oerror', {one: 'one', two: 2}))
---
{
   "name":"spelling",
   "hostname":"3e10c4539405",
   "pid":18,
   "level":30,
   "err":{
      "message":"message oerror",
      "name":"OError",
      "stack":"OError: message oerror\n    at Object.<anonymous> (/app/app/js/ASpell.js:19:12)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIManager.js:9:16)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIController.js:1:28)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app.js:19:31)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)\n    at startup (internal/bootstrap/node.js:283:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)",
      "info":{
         "one":"one",
         "two":2
      }
   },
   "msg":"message oerror",
   "time":"2020-06-04T11:46:42.133Z",
   "v":0
}
```

```
const error = new Error('inner message', {one: 'one'})
logger.error(OError.tag(error, 'outer message', {two: 'two'}))
---
{
   "name":"spelling",
   "hostname":"3e10c4539405",
   "pid":19,
   "level":50,
   "err":{
      "message":"inner message",
      "name":"Error",
      "stack":"Error: inner message\n    at Object.<anonymous> (/app/app/js/ASpell.js:19:15)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIManager.js:9:16)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIController.js:1:28)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app.js:19:31)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)\n    at startup (internal/bootstrap/node.js:283:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)",
      "info":{

      }
   },
   "msg":"undefined",
   "time":"2020-06-04T11:50:20.348Z",
   "v":0
}
```

```
const error = new OError('inner message', {one: 'one'})
logger.error(new OError('outer message', {two: 'two'}, error))
---
{
   "name":"spelling",
   "hostname":"3e10c4539405",
   "pid":18,
   "level":50,
   "err":{
      "message":"outer message",
      "name":"OError",
      "stack":"OError: outer message\n    at Object.<anonymous> (/app/app/js/ASpell.js:20:14)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIManager.js:9:16)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIController.js:1:28)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app.js:19:31)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)\n    at startup (internal/bootstrap/node.js:283:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)\ncaused by: OError: inner message\n    at Object.<anonymous> (/app/app/js/ASpell.js:19:15)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIManager.js:9:16)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIController.js:1:28)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app.js:19:31)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)\n    at startup (internal/bootstrap/node.js:283:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)",
      "info":{
         "one":"one",
         "two":"two"
      }
   },
   "msg":"undefined",
   "time":"2020-06-04T11:54:23.344Z",
   "v":0
}
```

```
const error = new OError('inner message', {one: 'one'})
logger.error(OError.tag(error, 'outer message', {two: 'two'}))
---
{
   "name":"spelling",
   "hostname":"ca2bc3af1efb",
   "pid":18,
   "level":50,
   "err":{
      "message":"inner message",
      "name":"OError",
      "stack":"OError: inner message\n    at Object.<anonymous> (/app/app/js/ASpell.js:19:15)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIManager.js:9:16)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app/js/SpellingAPIController.js:1:28)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Module.require (internal/modules/cjs/loader.js:692:17)\n    at require (internal/modules/cjs/helpers.js:25:18)\n    at Object.<anonymous> (/app/app.js:19:31)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)\n    at startup (internal/bootstrap/node.js:283:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)",
      "info":{
         "one":"one"
      }
   },
   "msg":"undefined",
   "time":"2020-06-04T11:56:51.614Z",
   "v":0
}
```

[the remark this answers is not in the document]
Comment (emcsween) [reply app/js/ASpell.js:33]: Sorry, my bad. I didn't know that worked. It's an exception for errors, and it's [documented](https://github.com/trentm/node-bunyan#log-method-api).

**Pull request merged**